sdk/eventgrid/eventgrid/src/eventGridClient.ts

```diff
@@ -119,61 +127,92 @@ export class EventGridPublisherClient {
    *
    * @param message One or more events to publish
    */
-  sendEvents(
+  async sendEvents(
     events: SendEventGridEventInput<any>[],
     options?: SendEventsOptions
   ): Promise<SendEventsResponse> {
-    return this.client
-      .publishEvents(
+    const { span, updatedOptions } = createSpan(
+      "EventGridPublisherClient-sendEvents",
+      options || {}
+    );
+
+    try {
+      const r = await this.client.publishEvents(
         this.endpointUrl,
         (events || []).map(convertEventGridEventToModelType),
-        options
-      )
-      .then((r) => {
-        return {
-          _response: r._response
-        };
-      });
+        updatedOptions
+      );
+      return {
+        _response: r._response
```

Comment:
`_response` generally shouldn't be enumerable, or something like JSON.serialize will happily serialize all your private auth headers in the request property.